src/bzfs/bzfsAPI.cxx

```diff
@@ -4286,10 +4286,29 @@ BZF_API const char* bz_getServerOwner()
     return getPublicOwner().c_str();
 }
 
+BZF_API void bz_reloadAll()
+{
+    bz_ReloadData_V1 event;
+    event.target = "all";
+    worldEventManager.callEvents(bz_eReloadEvent, &event);
+
+    bz_reloadLocalBans();
+    bz_reloadMasterBans();
+    bz_reloadGroups();
+    bz_reloadUsers();
+    bz_reloadHelp();
+    bz_reloadBadwords();
+}
+
 BZF_API void bz_reloadLocalBans()
 {
     // reload the banlist
     logDebugMessage(3,"Reloading bans\n");
+
+    bz_ReloadData_V1 event;
+    event.target = "bans";
+    worldEventManager.callEvents(bz_eReloadEvent, &event);
```

Comment:
as noted below, what's the proper interaction with the event manager? Should the call be before the `acl.load()` like this, meaning this might potentially undo a plugin action, or should the event manager call be after the defined action? Or should the event be tested to see if it was handled?